### PR TITLE
Watcher: Configure HttpClient parallel sent requests

### PR DESCRIPTION
**Pull request by spinscale**
The HTTPClient used in watcher is based on the apache http client. The
current client is using a lot of defaults - which are not always
optimal. Two of those defaults are the maximum number of total
connections and the maximum number of connections to a single route.

If one of those limits is reached, the HTTPClient waits for a connection
to be finished thus acting in a blocking fashion. In order to prevent
this when many requests are being executed, we increase the limit of
total connections as well as the connections per route (a route is
basically an endpoint, which also contains proxy information, not
containing an URL, just hosts).

On top of that an additional option has been made configurable to evict
long running connections, which can potentially be reused after some
time. As this requires an additional background thread, this required
some changes to ensure that the httpclient is closed properly. Also the
timeout for this can be configured.

Reviewers note: I am happy to discuss the naming of those options, I am not a fan of the implementation of the http client in the setting name, but if we ever change that client again (which we might once the built-in java HTTP client becomes usable), it might be easier to differentiate - I tend to lean to remove the `apache` though.

### CODE REVIEW

**Review by rjernst — COMMENTED** [review]
I think we should avoid the `apache` in the setting names, as you mentioned in your summary.

But why do we need these to be settable by users? Could we start by changing the values we use internally, and only adding the settings if users have a need to use the non defaults, if/when we get enough feedback asking for it?

**Comment by spinscale** [comment]
I think we can go without making them configurable at the beginning. I set the number of open connections to a pretty high number so people should not run into any limits with a 'normal' usage of watcher and left a comment.

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/LocalStateCompositeXPackPlugin.java [hunk 1]
```diff
@@ -391,6 +392,13 @@ public void onIndexModule(IndexModule indexModule) {
                 .collect(toList());
     }
 
+    @Override
+    public void close() throws IOException {
+        for (Plugin plugin : plugins) {
+            plugin.close();
```
Comment (rjernst) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/LocalStateCompositeXPackPlugin.java]: I think this should be `IOUtils.close(plugins);`?

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by spinscale** [comment]
@elasticmachine retest this please

**Pull request merged**